src/main/java/core/config/SecurityConfig.java

```diff
@@ -0,0 +1,38 @@
+package core.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    public void globalConfig(AuthenticationManagerBuilder managerBuilder) throws Exception {
+        managerBuilder.inMemoryAuthentication()
+                .withUser("user")
+                .password(getEncoder().encode("1234"))
+                .roles("USER");
+    }
+    
+    protected void configure(HttpSecurity security) throws Exception {
+        security.authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .formLogin().permitAll()
+                .and()
+                .httpBasic().and()
+                .csrf().disable();
+    }
```

Comment:
```suggestion
    protected void configure(HttpSecurity security) throws Exception {
        security.authorizeRequests()
        .antMatchers("/register")
                .permitAll()
                .anyRequest().authenticated()
                .and()
                .formLogin().permitAll()
                .and()
                .httpBasic().and()
                .csrf().disable();
    }
```
we can add this if we want enter into /register endpoint without credentials